encfs/main.cpp

```diff
@@ -575,6 +607,18 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
+  // Let's unmount if requested
+  if (encfsArgs->opts->unmount) {
+    fuse_unmount(encfsArgs->opts->mountPoint.c_str(), nullptr);
+#ifdef __APPLE__
+    // fuse_unmount does not work on Mac OS, see #428
+    unmount(encfsArgs->opts->mountPoint.c_str(), MNT_FORCE);
```

Comment:
Why MNT_FORCE?

You should check the return code of both fuse_unmount and unmount

Reply:
As written in a comment some lines above, `fuse_unmount` succeeds and returns void.
So we can't check its return code.
And as it succeeds, we must free the mount point.
On Mac OS `fuse_unmount` does not do all the job, leaving the FS in a stalled state. This is why we use an additional `unmount` with `MNT_FORCE`, to be sure to unmount.

Reply:
I confirm that this code (`unmount(encfsArgs->opts->mountPoint.c_str(), MNT_FORCE);`) is necessary on MacOS.

Reply:
@mouse07410 thanks for testing!